### PR TITLE
11 delete comments

### PR DESCRIPTION
**Pull request by muhammeddevops**

### CODE REVIEW

File: src/components/IndvReview.jsx [hunk 1]
```diff
@@ -24,7 +29,7 @@ export const IndvReview = () => {
     getCommentsOfReview(review_id).then((comments) => {
       setComments(comments.results);
     });
-  }, [review_id]);
+  }, [review_id, comments]);
```
Comment (dav3rid) [inline src/components/IndvReview.jsx]: Here is where the problem stems from

File: src/components/IndvReview.jsx [hunk 2]
```diff
@@ -35,6 +40,12 @@ export const IndvReview = () => {
     });
   };
 
+  const handleDelete = (comment_id) => {
+    deleteComment(comment_id).then((comment) => {
+      setComments(comments);
```
Comment (dav3rid) [inline src/components/IndvReview.jsx]: This is just setting the comments to what they were before and feeding into your problem with the other useEffect

File: src/components/IndvReview.jsx [hunk 3]
```diff
@@ -77,7 +88,18 @@ export const IndvReview = () => {
...
+                  <button
+                    type="button"
+                    onClick={() => {
+                      handleDelete(comment.comment_id);
+                    }}
+                  >
+                    delete
+                  </button>
+                ) : (
+                  <p></p>
```
Comment (dav3rid) [inline src/components/IndvReview.jsx]: Try not to have any empty tags present

**Review by dav3rid — APPROVED** [review]
You have an infinite loop running - looks like you have `comments` in a dependency array for the `useEffect` where you fetch them - you'll need to remove this

File: src/components/IndvReview.jsx [hunk 4]
```diff
@@ -35,6 +40,16 @@ export const IndvReview = () => {
     });
   };
 
+  const handleDelete = (comment_id) => {
+    deleteComment(comment_id).then(() => {
+      setComments((currComments) => {
+        const newComments = [...currComments];
+        newComments.shift();
+        return newComments;
+      });
```
Comment (Keb84) [inline src/components/IndvReview.jsx:49]: There seems to be something going on here that's affecting the delete. When i log in and try to delete a comment it's not deleting the right one and i'm getting an error in the console 👀

**Review by Keb84 — CHANGES_REQUESTED** [review]

**Pull request merged**